spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/mapping/DefaultNeo4jConverter.java

```diff
@@ -470,10 +471,15 @@ public Object getParameterValue(PreferredConstructor.Parameter parameter) {
 			for (Value relatedEntity : list.asList(Function.identity())) {
 				Neo4jPersistentProperty idProperty = concreteTargetNodeDescription.getRequiredIdProperty();
 
+				String relatedEntityIdKey = Optional.ofNullable(concreteTargetNodeDescription.getIdDescription())
+					.flatMap(IdDescription::getOptionalGraphPropertyName)
+					.orElse(idProperty.getName());
+
```

Comment:
```suggestion
				// internal (generated) id or external set
				String relatedEntityIdKey = idProperty.isInternalIdProperty() ? NAME_OF_INTERNAL_ID : concreteTargetNodeDescription.getIdDescription()
					.getOptionalGraphPropertyName()
					.orElse(idProperty.getName());

```

If there's an id property, then there's also an id description, so the thing cannot be null. As this is a rather hot path, I'd like to try to avoid the `Optional` creation (which is also the reason behind pulling the condition (whether internal or external) up.